LayoutTests/imported/w3c/web-platform-tests/css/css-grid/parsing/grid-shorthand-serialization.html

```diff
@@ -0,0 +1,137 @@
+<meta name="assert" content="grid shorthand must not serialize when the value cannot roundtrip.">
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+</head>
+<body>
+<script>
+  function test_shorthand_roundtrip(cssText, property, value, longhands, declarations) {
+    var div = document.createElement('div');
+    div.style.cssText = cssText;
+
+    const should = (value === "") ? "; shouldn't" : "; should";
+    test(function(){
+      try {
+        const readValue = div.style[property];
+        if (Array.isArray(value)) {
+          assert_in_array(readValue, value);
+        } else {
+          assert_equals(readValue, value);
+        }
+      } finally {}
+    }, "e.style.cssText = " + cssText + should + " serialize the " + property + " shorthand");
+
+    if (longhands) {
+      for (let longhand of Object.keys(longhands).sort()) {
+        test(function(){
+          try {
+            const readValue = div.style[longhand];
+            assert_equals(readValue, longhands[longhand], longhand + " should be canonical");
+
+            div.style[longhand] = "";
+            div.style[longhand] = readValue;
+            assert_equals(div.style[longhand], readValue, "serialization should round-trip");
+          } finally {}
+        }, "e.style.cssText = " + cssText + " should set " + longhand);
+      }
+    }
+
+    if (declarations) {
+      let cssText = div.style.cssText;
+      declarations.forEach(decl => {
+        try {
+          assert_true(cssText.indexOf(decl) !== -1, `cssText ('${cssText}') must contain '${decl}'`);
+        } finally {}
+      });
+    }
+  }
+
+  test_shorthand_roundtrip('grid: auto-flow auto / 100px 100px',
+                            'grid', ['auto-flow / 100px 100px', 'none / 100px 100px'],
+                            {
+    'grid-template-areas': 'none'
+  });
+  
+  test_shorthand_roundtrip('grid: auto-flow auto / 100px 100px; grid-template-areas: "one two" "three four"',
+                            'grid', '',
+                            {
+    'grid-template-areas': '"one two" "three four"',
+    'grid-auto-flow': 'row',
+    'grid-auto-rows': 'auto'
+  });
+
+  test_shorthand_roundtrip('grid: 30px 40px / 50px 60px; grid-auto-flow: column',
+                           'grid', '',
+                           {
+    'grid-template': '30px 40px / 50px 60px',
+    'grid-auto-flow': 'column',
+  }, [
+    'grid-template: 30px 40px / 50px 60px;',
+    'grid-auto-rows: auto;',
+    'grid-auto-columns: auto;',
+    'grid-auto-flow: column;'
+  ]);
+
+  test_shorthand_roundtrip('grid: auto-flow / 10px; grid-template-rows: 20px',
+                           'grid', '20px / 10px',
+                           {
+    'grid-template': '20px / 10px'
+  }, [
+    'grid-template: 20px / 10px;',
+    'grid-auto-flow: row;',
+    'grid-auto-columns: auto;',
+    'grid-auto-rows: auto;',
+  ]);
```

Comment:
This was already wrong, but if `grid` serializes, then `cssText` should just be `grid: 20px / 10px`?